pkgs/by-name/lm/lm4flash/package.nix

```
@@ -0,0 +1,50 @@
+{
+  fetchFromGitHub,
+  stdenv,
+  libusb1,
+  pkg-config,
+  lib,
+  writeText,
+}:
+let
+  version = "0.1.3";
+  lm4tools = fetchFromGitHub {
+    owner = "utzig";
+    repo = "lm4tools";
+    rev = "v${version}";
+    sha256 = "sha256-ZjuCH/XjQEgg6KHAvb95/BkAy+C2OdbtBb/i6K30+uo=";
+  };
+  stellaris-udev-rules = writeText "61.stellpad.rules" ''
+    SUBSYSTEM=="usb", ATTRS{idVendor}=="1cbe", ATTRS{idProduct}=="00fd", MODE="0666"
+  '';
+in
+stdenv.mkDerivation rec {
+  pname = "lm4flash";
+  inherit version;
+  src = "${lm4tools}/lm4flash";
+  buildInputs = [
+    libusb1
+    pkg-config
+  ];
+  buildPhase = ''
+    make release
+  '';
+  installPhase = ''
+    make install PREFIX=$out
+    mkdir -p "$out/etc/udev/rules.d"
+    install -m644 "${stellaris-udev-rules}" "$out/etc/udev/rules.d/61.stellpad.rules"
+  '';
```

For better or worse, the default behaviour of mkDerivation tries to work with `make`, so there's no need to manually call the `make` commands.
Also, you can use `-D` to automatically create the directory when using `install`

```suggestion
  buildFlags = [ "release" ];

  installFlags = [ "PREFIX=$(out)" ];

  postInstall = ''
    install -Dm644 "${stellaris-udev-rules}" "$out/etc/udev/rules.d/61.stellpad.rules"
  '';
```